cmd/package-operator-manager/main.go

```diff
@@ -77,7 +79,17 @@ func run(opts components.Options) error {
 			return err
 		}
 
-		if err := bs.Bootstrap(ctx, func(ctx context.Context) error {
+		if err := bs.Bootstrap(ctx, func(bootstrapCtx context.Context) error {
+			_, err := logr.FromContext(bootstrapCtx)
+			if err != nil {
+				// Only get logger from original context and add it to bootstrapCtx if bootstrapCtx doesn't have one
+				logger, err := logr.FromContext(ctx)
+				if err != nil {
+					return fmt.Errorf("getting logger from context in bootstrap function: %w", err)
+				}
+				bootstrapCtx = logr.NewContext(bootstrapCtx, logger)
```

Comment:
This procedure is not necessary because bs.Bootstrap calls the callback with a child of the context that is passed in.

I'd argue that this should just return `err` when it's not nil.